witness/golang/omniwitness/usbarmory/internal/storage/slots/journal.go

```diff
@@ -102,12 +106,13 @@ func (e *entry) Size() int {
 // OpenJournal returns a new journal structure for interacting with a journal stored in the
 // [start, start+length) range of blocks accessible via dev.
 // Journal ranges should not overlap with one another, or corruption will almost certainly occur.
-func OpenJournal(dev BlockReaderWriter, start, length uint) (*Journal, error) {
+func OpenJournal(dev BlockReaderWriter, start, length uint, h Hasher) (*Journal, error) {
```

Comment:
It's obvious, but I think worth noting in the comment that the hasher passed in must functionally match any previous hasher used in this journal, or else everything will look corrupted!

Reply:
I've just renamed to make it clear this is always just SHA256 (but allowing for different implementations of it)

Reply:
Hmmm, I guess that works. Any validation scheme would work as long as it was consistent, but if you nail it down to sha256 then it's always consistent.